chainstate/test-suite/src/tests/chainstate_accounting_storage_tests.rs

```diff
@@ -0,0 +1,677 @@
+            pool_data: BTreeMap::from([(
+                pool_id1,
+                PoolData::new(pub_key.clone(), amount_to_stake),
+            )]),
+            pool_balances: BTreeMap::from([(pool_id1, amount_to_stake)]),
+            delegation_balances: Default::default(),
+            delegation_data: Default::default(),
+            pool_delegation_shares: Default::default(),
+        };
+        assert_eq!(
+            storage.read_accounting_data_sealed().unwrap(),
+            expected_sealed_storage_data
+        );
+
+        // check that deltas per block are stored
+        let pool_data = PoolData::new(pub_key.clone(), Amount::from_atoms(100));
+        let expected_block1_delta = pos_accounting::PoSAccountingDeltaData {
+            pool_data: DeltaDataCollection::from_iter(
+                [(pool_id1, DataDelta::new(None, Some(pool_data.clone())))].into_iter(),
+            ),
+            pool_balances: DeltaAmountCollection::from_iter(
+                [(pool_id1, amount_to_stake.into_signed().unwrap())].into_iter(),
+            ),
+            pool_delegation_shares: DeltaAmountCollection::new(),
+            delegation_balances: DeltaAmountCollection::new(),
+            delegation_data: DeltaDataCollection::new(),
+        };
+
+        let block1_delta = storage
+            .get_accounting_delta(*block1_index.block_id())
+            .expect("ok")
+            .expect("some");
+        assert_eq!(block1_delta, expected_block1_delta);
+
+        let expected_block2_delta = pos_accounting::PoSAccountingDeltaData {
+            pool_data: DeltaDataCollection::from_iter(
+                [(pool_id2, DataDelta::new(None, Some(pool_data)))].into_iter(),
+            ),
+            pool_balances: DeltaAmountCollection::from_iter(
+                [(pool_id2, amount_to_stake.into_signed().unwrap())].into_iter(),
+            ),
+            pool_delegation_shares: DeltaAmountCollection::new(),
+            delegation_balances: DeltaAmountCollection::new(),
+            delegation_data: DeltaDataCollection::new(),
+        };
+
+        let block2_delta = storage
+            .get_accounting_delta(*block2_index.block_id())
+            .expect("ok")
+            .expect("some");
+        assert_eq!(block2_delta, expected_block2_delta);
+
+        assert!(storage.get_accounting_epoch_undo_delta(0).unwrap().is_none());
+        assert!(storage.get_accounting_epoch_undo_delta(1).unwrap().is_some());
+        assert!(storage.get_accounting_epoch_undo_delta(2).unwrap().is_none());
+    });
+}
+
+// Config chain to seal an epoch every block and the distance between tip and sealed to 0
+// (meaning every block is sealed thus tip == sealed).
+// Create block1 from genesis that creates a stake pool.
+// Check that the info is stored to the tip and sealed storage.
+// Check that delta from block is stored.
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn accounting_storage_seal_every_block(#[case] seed: Seed) {
```

Comment:
Are there tests for unsealing blocks, and going back and forth?

Reply:
There are tests for reorg with different lengths and distances in `chainstate/test-suite/src/tests/pos_accounting_reorg.rs`

As for unsealing there are no explicit tests. Chainstate interface doesn't expose disconnect functionality, so I'm not sure how to approach it.